backend/app/channels/game_channel.rb

```diff
@@ -0,0 +1,22 @@
+class GameChannel < ApplicationCable::Channel
+  def subscribed
+    stream_from "game_channel_#{current_user.id}"
+  end
+  def unsubscribed
+    Game.leave_game(current_user) if current_user.present?
```

Comment:
Instead of using a custom method, it would be cleaner to do something descriptive like this:

```ruby
Game.active.played_by(current_user).destroy_all
```

Reply:
Is it possible that there is no `current_user` here?

Reply:
yes, If the user clears cookies of the browser, Or in future, we plan to give a logout button on the game page and there is a scenario that user can log out before unsubscribing.